### ISSUE TITLE
Wcaro Togo - Data Sync Issues

### ISSUE THREAD

**Issue by ageryck** [opensrp/opensrp-client-chw#1908]
As a result of the current Togo Upgrade, one region reported sync issues once again. Majority of devices from `Kpendjal` were not able to successfully sync to completion (as communicated from client out of estimated 150 devices only 6 were successfully synced) this is very compared to similar exercise conducted in `Bassar` region where out of an estimate of 300 devices over 200 very synced successfully. 

On Friday 24/09/2021, we conducted live sessions with client on about 9 devices and only 2 of those were able to sync to completion. The devices belonged to the following providers;
```
--kdouti08
--dsafaguidja33
--fgountante91
--glare85
--mlamboni96
--mkome00
--bkombate95
--aarouna52
--slamboni07
```
During the live session we noticed Togo Prod Instance throwing error 499 for some few requests, we did an apk/server fix by increasing timeout to 5 minutes on server-side and reducing the batch size from 250 to 180. We engaged the client after that and the results indicated no more 499 however the sync status of selected 7 devices left overnight on wifi only 1 was successfully synced come morning. 

Something else worth investigating is that with all the above sampled devices a comparison of extracted [DB extracts](https://drive.google.com/drive/folders/1ABIZa7HxhH4K6a3agEuPY13a1nCRFl1w?usp=sharing) vs data on OpenSRP server showed records match in majority of them after sync (failed synced), a closer look on one (`mlamboni96`) from the sync stats page it synced everything to `0` unsynced status then restarted the process again and never ending. This seemed to be the case for the devices we tested with. Question is why wouldn't synced indicated completion once all records and uploaded to servers and/or why would the devices restart sync process again after syncing everything ?

**Comment by ageryck** [comment]
We made server side updated to optimize query process time and make server request processing more efficient, we are awaiting client focal point in Kpendjal to retest sync of the said devices.
Additionally we have determined that for these specific devices may have been not updated with some intermediary versions that eliminated a local device database bug, in that sense we are proposing of sharing a new apk with client to help clean up local database issues for devices to be able to sync successfully.
We determined that there exist a column on local DB named `eventId` that is updated after successful sync, an additional check that runs to invalidate the wrongly validated event would check if this field is `null` and invalidate those records. It turned out that for some devices the field was always `null` hence the issues of sync always restarting.
